docs/topics/multiplatform-mobile/multiplatform-mobile-upgrade-app.md

```diff
@@ -0,0 +1,348 @@
+                })
+            }
+        }
+    }
+    ```
+
+    To deserialize the result of the GET request,
+    the [ContentNegotiation Ktor plugin](https://ktor.io/docs/serialization-client.html#register_json) and the JSON
+    serializer are used.
+
+2. In the `greeting()` function, retrieve the information about rocket launches by calling the `httpClient.get()`
+   method and find the latest one:
+
+    ```kotlin
+    class Greeting {
+        // ...
+        @Throws(Exception::class)
+        suspend fun greeting(): String {
+            val rockets: List<RocketLaunch> = httpClient.get("https://api.spacexdata.com/v3/launches").body()
+            val firstSuccessLaunch = rockets.first { it.launchSuccess == true }
+            return "Guess what it is! > ${Platform().platform.reversed()}!" +
+                "\nThere are only ${daysUntilNewYear()} left! 🎅🏼 " +
+                "\nThe first success launch was ${firstSuccessLaunch.launchDateUTC} 🚀"
+        }
+    }
+    ```
+
+    The `suspend` modifier in the `greeting()` function is necessary because it now contains a call to `get()`. It's a
+    a suspend function that has an asynchronous operation to retrieve data over the internet and can only be called from
+    within a coroutine or another suspend function. The network request will be executed in the HTTP client's thread pool.
+
+### Add internet access permission
+
+To access the internet, the Android application needs appropriate permission. Since all network requests are made from the
+shared module, it makes sense to add the internet access permission to its manifest.
+
+Update your `shared/src/androidMain/AndroidManifest.xml` file the following way:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+package="com.jetbrains.onboarding" >
+<uses-permission android:name="android.permission.INTERNET"/>
+</manifest>
+```
+
+## Update Android and iOS apps
+
+You've already updated the API of our shared module by adding the `suspend` modifier to the `greeting()` function. Now you
+need to update native (iOS, Android) parts of the project, so they can properly handle the result of calling the
+`greeting()` function.
+
+### Android app
+
+As both a shared module and an Android application are written in Kotlin, using shared code from Android is
+straightforward:
+
+1. Add `kotlinx.coroutines` library to the Android application by adding a line in the `build.gradle.kts` in the
+   `androidApp` folder:
+
+    ```kotlin
+    dependencies {
+        // ..
+        implementation("org.jetbrains.kotlinx:kotlinx-coroutines-android:1.6.2")
+    }
+    ```
+
+2. In `androidApp/src/main`, update the `MainActivity` class replacing previous implementation:
+
+    ```kotlin
+   import kotlinx.coroutines.MainScope
+   import kotlinx.coroutines.launch
+
+    class MainActivity : AppCompatActivity() {
+        private val mainScope = MainScope()
```

Comment:
In the MainScope docs, the recommended usage pattern is the following:

class MyAndroidActivity {
    private val scope = MainScope()

    override fun onDestroy() {
        super.onDestroy()
        scope.cancel()
    }
}

Hopefully, this guide soon converts to Compose, but before that, it's probably worth following the guidelines and adding the onDestroy function.